frontend/src/components/featured-post-card.tsx

```diff
@@ -9,17 +9,17 @@ export default function FeaturedPostCard({ post }: { post: Post }) {
   const slug = createSlug(post.title);
   return (
     <div
-      className="flex h-48 cursor-pointer gap-2 rounded-lg bg-light dark:bg-dark-card"
+      className="flex flex-col sm:flex-row h-auto sm:h-48 cursor-pointer gap-2 rounded-lg bg-light dark:bg-dark-card"
       onClick={() => navigate(`/details-page/${slug}/${post._id}`, { state: { post } })}
     >
-      <div className="w-1/3">
+      <div className="w-full sm:w-1/3">
         <img
           src={post.imageLink}
           alt={post.title}
-          className="h-full w-full rounded-lg object-cover shadow-lg"
+          className="h-48 sm:h-full w-full rounded-lg object-cover shadow-lg"
         />
```

Comment:
h-48 looks good to me, just check all the scenarios once again, to make sure it isn't looking odd.